### ISSUE TITLE
feat: ScaNN index

### ISSUE THREAD

**Issue by gaocegege** [tensorchord/pgvecto.rs#20]

**Comment by karajan1001** [comment]
I'll take this.

**Comment by VoVAllen** [comment]
@karajan1001 ScaNN might be hard to implement since original code highly coupled with tensorflow. I would suggest DiskANN, since it's designed for disk workload, which is suitable for postgres. And it's also adapted by Milvus.

**Comment by karajan1001** [comment]
> @karajan1001 ScaNN might be hard to implement since original code highly coupled with tensorflow. I would suggest DiskANN, since it's designed for disk workload, which is suitable for postgres. And it's also adapted by Milvus.

Yes, I had looked into their paper and source code and noticed that it's highly coupled with TensorFlow. So it's hard to say how much performance is gotten from the algorithm and how much is gotten from the TensorFlow optimization.

**Comment by gaocegege** [comment]
https://services.google.com/fh/files/misc/scann_for_alloydb_whitepaper.pdf

Here is a whitepaper for SCANN in AlloyDB.
